frontend/lib/resolution/Resolver.cpp

```diff
@@ -394,7 +394,16 @@ const Scope* Resolver::methodReceiverScope(bool recompute) {
       auto* untyped = typedSignature->untyped();
       auto thisFormal = untyped->formalDecl(0)->toFormal();
       auto type = thisFormal->typeExpression();
-      if (auto ident = type->toIdentifier()) {
+      if (type == nullptr) {
+        // `this` formals of primary methods have no type expression. They are,
+        // however, in primary methods, so the method's parent is the aggregate
+        // type whose scope should be used.
+
+        // TODO can typedSignature->id() ever not refer to a Function in this case?
+        auto recordOrClassId = parsing::idToParentId(context,
```

Comment:
Here's a question I don't know the answer to: can you have nested methods?

Reply:
I think you can declare a function inside a method, but then it's a function (and presumably gets lifted?). But I'd better check.

Reply:
Nested function declarations are allowed, but they don't get an implicit this formal, so this shouldn't pose a problem for this PR.